sql/delete.go

```diff
@@ -39,7 +39,7 @@ func (p *planner) Delete(n *parser.Delete) (planNode, error) {
 	// deleting.
 	node, err := p.Select(&parser.Select{
 		Exprs: parser.SelectExprs{
-			&parser.StarExpr{TableName: parser.QualifiedName{tableDesc.Name}},
+			&parser.StarExpr{TableName: &parser.QualifiedName{Base: parser.Name(tableDesc.Name)}},
```

Comment:
does the distinction between `StarExpr` and `NonStarExpr` still make sense? In the grammar changes below, you use the `StarIndirection` for `something.*` but as far as I can tell outside of a `StarExpr`. Would `*` give you something else?